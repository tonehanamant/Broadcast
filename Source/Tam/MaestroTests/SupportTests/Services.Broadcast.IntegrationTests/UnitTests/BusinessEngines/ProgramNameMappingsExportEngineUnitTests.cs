using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;
using Services.Broadcast.Entities;
using System;
using System.Collections.Generic;

namespace Services.Broadcast.IntegrationTests.UnitTests.BusinessEngines
{
    [TestFixture]
    public class ProgramNameMappingsExportEngineUnitTests
    {
        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void GetColumnDescriptors()
        {
            
            var engine = new ProgramNameMappingsExportEngineUnitTestClass(null,null);

            var result = engine.UT_GetColumnDescriptors();

            Approvals.Verify(IntegrationTestHelper.ConvertToJson(result));
        }
        
        [Test]
        [UseReporter(typeof(DiffReporter))]         
        public void UT_GetColumnDescriptorsWithoutGenres()
        {
            var engine = new ProgramNameMappingsExportEngineUnitTestClass(null, null);
            var result = engine.UT_GetColumnDescriptorsWithoutGenres();
            Approvals.Verify(IntegrationTestHelper.ConvertToJson(result));
        }

        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void GetDateGeneratedCellValue()
        {
            var engine = new ProgramNameMappingsExportEngineUnitTestClass(null, null)
            {
                UT_CurrentDateTime = new DateTime(2020, 10, 17, 7, 31, 26)
            };

            var result = engine.UT_GetDateGeneratedCellValue();

            Approvals.Verify(IntegrationTestHelper.ConvertToJson(result));
        }

        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void TransformToExportLines_CreatedDates()
        {
            var createDateTime = new DateTime(2020, 10, 17, 7, 31, 26);
            DateTime? modifiedDateTime = null;
            const int testProgramCount = 10;
            var mappings = _GetTestPrograms(testProgramCount, createDateTime, modifiedDateTime);
            var engine = new ProgramNameMappingsExportEngineUnitTestClass(null, null)
            {
                UT_CurrentDateTime = createDateTime
            };

            var result = engine.UT_TransformToExportLines(mappings);

            Approvals.Verify(IntegrationTestHelper.ConvertToJson(result));
        }

        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void TransformToExportLines_ModifiedDates()
        {
            var createDateTime = new DateTime(2020, 10, 17, 7, 31, 26);
            DateTime? modifiedDateTime = new DateTime(2020, 10, 20, 7, 31, 26);
            ;
            const int testProgramCount = 10;
            var mappings = _GetTestPrograms(testProgramCount, createDateTime, modifiedDateTime);
            var engine = new ProgramNameMappingsExportEngineUnitTestClass(null, null)
            {
                UT_CurrentDateTime = createDateTime
            };

            var result = engine.UT_TransformToExportLines(mappings);

            Approvals.Verify(IntegrationTestHelper.ConvertToJson(result));
        }

        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void TransformToExportLines_WithoutGenre()
        {
            var createDateTime = new DateTime(2020, 10, 17, 7, 31, 26);
            DateTime? modifiedDateTime = new DateTime(2020, 10, 20, 7, 31, 26);
            ;
            const int testProgramCount = 10;
            var mappings = _GetTestPrograms(testProgramCount, createDateTime, modifiedDateTime);
            var engine = new ProgramNameMappingsExportEngineUnitTestClass(null, null)
            {
                UT_CurrentDateTime = createDateTime
            };

            var result = engine.UT_TransformToExportLinesWithoutGenres(mappings);                

            Approvals.Verify(IntegrationTestHelper.ConvertToJson(result));
        }

        private List<ProgramMappingsDto> _GetTestPrograms(int count, DateTime createdDateTime, DateTime? modifiedDateTime)
        {
            var programs = new List<ProgramMappingsDto>();
            for (var i = 0; i < count; i++)
            {
                var program = new ProgramMappingsDto
                {
                    Id = 1,
                    OriginalProgramName = $"OriginalProgramName{i}",
                    OfficialProgramName = $"OfficialProgramName{i}",
                    OfficialGenre = new Genre {Name = $"OfficialGenre{i}"},
                    OfficialShowType = new ShowTypeDto {Name = $"OfficialShowType{i}"},
                    CreatedBy = "TestUser",
                    CreatedAt = createdDateTime,
                    ModifiedBy = modifiedDateTime.HasValue ? "TestUser" : null,
                    ModifiedAt = modifiedDateTime
                };
                programs.Add(program);
            }
            return programs;
        }
    }
}