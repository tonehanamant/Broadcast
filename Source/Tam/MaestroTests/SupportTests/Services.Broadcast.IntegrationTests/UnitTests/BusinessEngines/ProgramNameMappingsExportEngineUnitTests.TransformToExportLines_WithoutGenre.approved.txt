[
  [
    "OriginalProgramName0",
    "OfficialProgramName0",
    "OfficialShowType0",
    "TestUser",
    "2020-10-20T07:31:26"
  ],
  [
    "OriginalProgramName1",
    "OfficialProgramName1",
    "OfficialShowType1",
    "TestUser",
    "2020-10-20T07:31:26"
  ],
  [
    "OriginalProgramName2",
    "OfficialProgramName2",
    "OfficialShowType2",
    "TestUser",
    "2020-10-20T07:31:26"
  ],
  [
    "OriginalProgramName3",
    "OfficialProgramName3",
    "OfficialShowType3",
    "TestUser",
    "2020-10-20T07:31:26"
  ],
  [
    "OriginalProgramName4",
    "OfficialProgramName4",
    "OfficialShowType4",
    "TestUser",
    "2020-10-20T07:31:26"
  ],
  [
    "OriginalProgramName5",
    "OfficialProgramName5",
    "OfficialShowType5",
    "TestUser",
    "2020-10-20T07:31:26"
  ],
  [
    "OriginalProgramName6",
    "OfficialProgramName6",
    "OfficialShowType6",
    "TestUser",
    "2020-10-20T07:31:26"
  ],
  [
    "OriginalProgramName7",
    "OfficialProgramName7",
    "OfficialShowType7",
    "TestUser",
    "2020-10-20T07:31:26"
  ],
  [
    "OriginalProgramName8",
    "OfficialProgramName8",
    "OfficialShowType8",
    "TestUser",
    "2020-10-20T07:31:26"
  ],
  [
    "OriginalProgramName9",
    "OfficialProgramName9",
    "OfficialShowType9",
    "TestUser",
    "2020-10-20T07:31:26"
  ]
]